packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -58,94 +83,102 @@ const useStyles = makeStyles(({ spacing, typography, palette }) => ({
     { id: trackId },
     { disabled: !trackId }
   )
+  const isUSDCEnabled = useIsUSDCEnabled()
+  const error = useSelector(getPurchaseContentError)
+  const stage = useSelector(getPurchaseContentFlowStage)
+  const isPurchaseSuccessful = stage === PurchaseContentStage.FINISH
   const { premium_conditions: premiumConditions } = track ?? {}
-  if (!track || !isPremiumContentUSDCPurchaseGated(premiumConditions))
+
+  const handleClosed = useCallback(() => {
+    dispatch(purchaseContentActions.cleanup())
+  }, [dispatch])
+
+  if (
+    !track ||
+    !isPremiumContentUSDCPurchaseGated(premiumConditions) ||
+    !isUSDCEnabled
+  )
     return null
   const price = formatPrice(premiumConditions.usdc_purchase.price)
 
   return (
-    <NativeDrawer drawerName={PREMIUM_TRACK_PURCHASE_MODAL_NAME}>
+    <NativeDrawer
+      drawerName={PREMIUM_TRACK_PURCHASE_MODAL_NAME}
+      onClosed={handleClosed}
+    >
       <View style={styles.drawer}>
         <View style={styles.titleContainer}>
           <IconCart fill={neutralLight2} />
           <Text style={styles.title}>{messages.title}</Text>
         </View>
         <TrackDetailsTile trackId={track.track_id} />
-        <View style={styles.summaryContainer}>
-          <View style={[styles.summaryRow, styles.greyRow]}>
-            <Text
-              weight='bold'
-              textTransform='uppercase'
-              style={styles.summaryTitle}
-            >
-              {messages.summary}
-            </Text>
-          </View>
-          <View style={styles.summaryRow}>
-            <Text>{messages.artistCut}</Text>
-            <Text>{messages.price(price)}</Text>
-          </View>
-          <View style={styles.summaryRow}>
-            <Text>{messages.audiusCut}</Text>
-            <Text>{messages.alwaysZero}</Text>
-          </View>
-          <View style={[styles.summaryRow, styles.lastRow, styles.greyRow]}>
-            <Text weight='bold'>{messages.youPay}</Text>
-            <Text weight='bold' color='secondary'>
-              {messages.price(price)}
-            </Text>
-          </View>
-        </View>
-        <View>
-          <View style={styles.payToUnlockTitleContainer}>
-            <Text weight='heavy' textTransform='uppercase' fontSize='small'>
-              {messages.payToUnlock}
+        <PurchaseSummaryTable
+          price={price}
+          isPurchaseSuccessful={isPurchaseSuccessful}
+        />
+        {isPurchaseSuccessful ? (
+          <PurchaseSuccess />
+        ) : (
+          <>
+            <View>
+              <View style={styles.payToUnlockTitleContainer}>
+                <Text weight='heavy' textTransform='uppercase' fontSize='small'>
+                  {messages.payToUnlock}
+                </Text>
+                <LockedStatusBadge locked />
+              </View>
+              <Text>
+                {messages.disclaimer(
+                  <Text
+                    onPress={() => console.log('terms of use pressed')}
```

Comment:
Planning to hook this up later? There were already existing places on web that linked out to terms of use which I was able to reuse.